spec/grape/api/custom_validations_spec.rb

```diff
@@ -162,9 +162,15 @@ def validate(request)
           return unless request.params.key? @attrs.first
           # check if admin flag is set to true
           return unless @option
+
           # check if user is admin or not
           # as an example get a token from request and check if it's admin or not
-          raise Grape::Exceptions::Validation.new(params: @attrs, message: 'Can not set Admin only field.') unless request.headers['X-Access-Token'] == 'admin'
+          header = if Gem::Version.new(Rack.release) < Gem::Version.new('3')
+                     'X-Access-Token'
+                   else
+                     'x-access-token'
+                   end
+          raise Grape::Exceptions::Validation.new(params: @attrs, message: 'Can not set Admin only field.') unless request.headers[header] == 'admin'
```

Comment:
Would it make more sense if `request.headers` became a dict with case-insensitive keys instead of downcasing everything?